docs/architecture/ADR-008-borrow-tracking.md

```diff
@@ -0,0 +1,81 @@
+# ADR 008: Borrow Tracking
+
+## Changelog
+
+- Jan 27, 2021: Initial Draft (@toteki)
+
+## Status
+
+Proposed
+
+## Context
+
+One of the more computationally expensive operations in Umee is "iterate over all borrows". It is currently necessary on several occasions:
+
+- When calculating borrow utilization (require's a denomination's total borrowed)
+- When accruing interest (dynamic interest requires borrow utilization, and all borrows must be modified)
+- When deriving uToken exchange rates (require's a denomination's total borrowed)
+
+In order to reduce the perofrmance impact of such iteration, `InterestEpoch` was created, so that such calculations only occurred every N blocks (e.g. 100).
```

Comment:
```suggestion
In order to reduce the performance impact of such iteration, `InterestEpoch` was created, so that such calculations only occurred every N blocks (e.g. 100).
```